src/contracts/core/DelegationManager.sol

```diff
@@ -340,24 +341,38 @@ contract DelegationManager is
      *          1) new delegations are not paused (PAUSED_NEW_DELEGATION)
      */
     function _delegate(address staker, address operator) internal onlyWhenNotPaused(PAUSED_NEW_DELEGATION) {
+        // read staker's withdrawable shares and strategies to add to operator's shares
+        // also update the staker depositScalingFactor for each strategy
+        (IStrategy[] memory strategies,) = strategyManager.getDeposits(staker);
+        uint256 podOwnerShares = eigenPodManager.stakerDepositShares(staker, beaconChainETHStrategy);
+        if (podOwnerShares != 0) {
+            // Allocate extra space for beaconChainETHStrategy and shares
+            IStrategy[] memory newStrategies = new IStrategy[](strategies.length + 1);
+            newStrategies[strategies.length] = beaconChainETHStrategy;
+
+            // Copy any strategy manager shares to complete array
+            for (uint256 i = 0; i < strategies.length; i++) {
+                newStrategies[i] = strategies[i];
+            }
+            strategies = newStrategies;
+        }
```

Comment:
Can we not just turn this whole thing into `(IStrategy[] memory strategies, ) = getDepositedShares(staker)`?

Saves half a kb

Reply:
it's definitely at a premium -- the other code is only saving maybe a few hundred gas at most, which is pennies compared to the bytecode space we have left. 100% we are optimizing for codesize, even if we spend more gas.